feat: [WLEO-320,WLEO-321] Upgraded native dependencies and exposed new `decodeIssuerSigned` method

sebastianopriscan opened the pull request:
## Short description

Upgraded versions of `io-react-native-cbor`'s native dependencies and exposed the new functionalities the dependecies brought:

More precisely, the library now exports a new method, `decodeIssuerSigned`, that, taken an `mDoc-CBOR`, decodes its `IssuerSigned` into json format, including a full decoding of its `IssuerAuth` (the raw `IssuerAuth` value is still present as a `raw` attribute of the `IssuerAuth` object).

## List of changes proposed in this pull request

- `pagopa-io-react-native-cbor.podspec` : upgraded `IOWalletCBOR` native dependency to `0.0.8` and restored the minimum supported ios platform supported to the default one
- `android/build.gradle` : upgraded `it.pagopa.io.wallet.cbor:cbor` dependency to `1.1.0`
- `example/ios/Podfile` : restored the minimum ios platform supported to the default one
- `example/ios/Podfile.lock` : new lock file for new dependencies
- `android/src/main/.../IoReactNativeCborModule.kt`, `ios/IoReactNativeCbor.mm`, `ios/IoReactNativeCbor.swift` : Exposed the method `decodeIssuerSigned` to allow decoding a single document `CBOR` with its `IssuerAuth` decoded both in Android and iOS
- `src/cbor/schema.ts` : Extended `IssuerSigned.issuerAuth` schema definition to adapt to changes in the native implementations which now decode the IssuerAuth
- `src/cbor/(decoder|index).ts` : Exposed new functions that map onto the new functionalities introduced by the native libraries
- `src/cbor/__tests__/schema.test.ts` : Evolved schema parsing test to adapt to schema changes
- `example/src/mocks/oneDocIssuerAuth.ts` : Mock `mDoc` with a single document
- `example/src/App.tsx` : Added new entry to test newly exposed native bridge functions


## How to test

0. Start from a clean build
    ``` bash
        cd example

        #iOS
        # If you store DeriveData with relative path to the project folder, otherwise use Xcode to clean the build
        rm -rf ios/DerivedData 
        rm -rf ios/Pods
        cd ios && bundler exec pod install && cd .. 
        yarn react-native run-ios

        #Android 
        cd android && ./gradlew/clean
        yarn react-native run-android
    ```
1. From the example app, try all the possible options, and assert that they terminate successfully.
2. From the project root, run `yarn test`

sebastianopriscan committed:
chore: [WLEO-320,WLEO-321] Upgraded versions of io-react-native-cbor's native dependencies
and exposed the new functionalities the dependecies brought
feat: [WLEO-320,WLEO-321] Upgraded versions of io-react-native-cbor's native dependencies
and exposed the new functionalities the dependecies brought

- pagopa-io-react-native-cbor.podspec : upgraded IOWalletCBOR native dependency to 0.0.8 and
  restored the minimum supported ios platform supported to the default one
- android/build.gradle : upgraded it.pagopa.io.wallet.cbor:cbor dependency to 1.1.0
- example/ios/Podfile : restored the minimum ios platform supported to the default one
- example/ios/Podfile.lock : new lock file for new dependencies
- android/src/main/.../IoReactNativeCborModule.kt, ios/IoReactNativeCbor.mm,
  ios/IoReactNativeCbor.swift : Exposed the method decodeIssuerSigned to
  allow decoding a single document CBOR with its IssuerAuth decoded both in Android and iOS
- src/cbor/schema.ts : Extended IssuerSigned.issuerAuth schema definition to adapt to changes
  in the native implementations which now decode the IssuerAuth
- src/cbor/(decoder|index).ts : Exposed new functions that map onto the new functionalities
  introduced by the native libraries
- src/cbor/__tests__/schema.test.ts : Evolved schema parsing test to adapt to schema changes
- example/src/mocks/oneDocIssuerAuth.ts : Mock mDoc with a single document
- example/src/App.tsx : Added new entry to test newly exposed native bridge functions

manuraf commented:
you could specify which functionalities you are adding...in this case you are exposing a new method `decodeIssuerSigned` that decode issuer signed in a json from a cbor

sebastianopriscan commented:
> you could specify which functionalities you are adding...in this case you are exposing a new method `decodeIssuerSigned` that decode issuer signed in a json from a cbor

Changed the short description and the title, let me know if the changes are valid

manuraf commented:
good

manuraf commented on ios/IoReactNativeCbor.swift:
```diff
@@ -43,6 +43,24 @@ class IoReactNativeCbor: NSObject {
   }
   
+  @objc func decodeIssuerSigned(
+    _ mdoc: String,
```
for clarity and consistency, you might consider having the parameter name match (e.g., data vs. mdoc, on Android function you have data) so that the bridging code looks symmetrical across platforms

sebastianopriscan replied:
The parameter names in both Android and iOS native bridges have been renamed `issuerSigned` with commit [d649217](https://github.com/pagopa/io-react-native-cbor/pull/4/commits/d64921701d034b850c3b97e3f1f5bd9c7ab7238c)

manuraf commented on example/ios/Podfile.lock:
```diff
@@ -7,12 +7,12 @@ PODS:
     - glog
     - hermes-engine
-    - IOWalletCBOR (~> 0.0.4)
+    - IOWalletCBOR (= 0.0.8)
```
is there any reason to pin exact versions instead of ~> ?....If you are comfortable tracking a minor-range version (e.g., ~> 0.0.8), you could allow patch-level updates automatically. Otherwise, be sure it’s pinned for a good reason

sebastianopriscan replied:
At the moment I tried to adjust dependencies this configuration seemed to be the only one to work, but I'm pretty sure it was more of a caching/syncronization issue. I'll try again using the minor range version starting from a clean build

sebastianopriscan replied:
With commit [d649217](https://github.com/pagopa/io-react-native-cbor/pull/4/commits/d64921701d034b850c3b97e3f1f5bd9c7ab7238c) this issue should have been solved, please try the change by starting from a clean build

manuraf commented on src/cbor/decoder.ts:
```diff
@@ -32,3 +37,19 @@ export const decodeDocuments = async (data: string): Promise<Documents> => {
+export const decodeIssuerSigned = async (
+  data: string
+): Promise<IssuerSigned> => {
+  const documentsString = await IoReactNativeCbor.decodeIssuerSigned(data);
+  return await IssuerSignedFromString.parseAsync(documentsString);
```
if you like to keep naming consistent, something like `issuerSignedString` rather than documentsString might read more clearly

sebastianopriscan replied:
With commit [d649217](https://github.com/pagopa/io-react-native-cbor/pull/4/commits/d64921701d034b850c3b97e3f1f5bd9c7ab7238c) the variable has been renamed along with the parameter received by the function, let me know if the new names are ok

sebastianopriscan committed:
chore: [WLEO-320,WLEO-321] Changed required IoWalletCBOR
iOS library version from "= 0.0.8" to "~> 0.0.8" and unified parameters' naming

- pagopa-io-react-native-cbor.podspec, example/ios/Podfile.lock :
  Changed IOWalletCBOR dependency
- src/cbor/decoder.ts : renamed generic "data" parameter to "issuerSigned",
  renamed "documentsString" variable to "decodedIssuerSignedString"
- android/src/.../IoReactNativeCborModule.kt, ios/IoReactNativeCbor.(mm|swift):
  unified string parameter name to "issuerSigned"

manuraf committed:
Merge branch 'main' into feat/WLEO-320-WLEO-321-update-native-dependencies-and-interfaces
